client/app/components/Modal/index.jsx

```diff
@@ -97,23 +98,53 @@ export class Modal extends React.Component<Props, State> {
     this.setState({ open: !open });
   };
 
-  render() {
+  resolveElement = () => {
     const { element, elementId } = this.props;
+    let renderComponent;
+
+    if (typeof element === 'object' && element.component) {
+      const { component, props } = element;
+      renderComponent = React.createElement(
+        this.resolveComponent(component),
+        { ...props },
+        null,
+      );
+    }
+
+    if (element) {
+      return (
+        <div
+          id={elementId || null}
```

Comment:
We can actually get rid of the null condition since elementId can be undefined or a string.